### DISCUSSION TITLE
pkcs12.toPkcs12Asn1 privateKey field has no null option in [@types/node-forge]

### DISCUSSION THREAD

**Discussion by andrewsomahony** [DefinitelyTyped/DefinitelyTyped#58453] [Issues with a @types package]
"null" needs to be an option for the key parameter in the pkcs12.toPkcs12Asn1 function

The current declaration for this function is:

function toPkcs12Asn1(
            key: pki.PrivateKey,
            cert: pki.Certificate | pki.Certificate[],
            password: string | null,
            options?: {
                algorithm?: 'aes128' | 'aes192' | 'aes256' | '3des' | undefined,
                count?: number | undefined,
                saltSize?: number | undefined,
                useMac?: boolean | undefined,
                localKeyId?: Hex | undefined,
                friendlyName?: string | undefined,
                generateLocalKeyId?: boolean | undefined,
            },
        ): asn1.Asn1;

However in the Javascript code, the privateKey is null checked, and if it is not null, it is added to the pkcs12 file, otherwise nothing is added.

[a comment by an automated account is not shown]

**Comment by beenotung** [comment]
Agree, I see a `  if(key !== null)` check in `node-forge/lib/pkcs12.js`

**Comment by beenotung** [comment]
Created a PR for this suggestion in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59032
